fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/block/type/BlockSetTypeRegistry.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.object.builder.v1.block.type;
+
+import net.minecraft.block.BlockSetType;
+import net.minecraft.sound.BlockSoundGroup;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.Identifier;
+
+/**
+ * This class allows registering {@link BlockSetType}s.
+ *
+ * <p>A {@link BlockSetType} is used to tell the game what sounds various related blocks should use.
+ *
+ * @see WoodTypeRegistry
+ */
+public final class BlockSetTypeRegistry {
+	private BlockSetTypeRegistry() {
+	}
+
+	/**
+	 * Creates and registers a {@link BlockSetType}.
+	 *
+	 * @param id the id of this {@link BlockSetType}
+	 * @return a new {@link BlockSetType}.
+	 */
+	public static BlockSetType registerBlockSetType(Identifier id) {
```

Comment:
This should be renamed; the block set created is specifically for overworld woods.